### PR TITLE
[Components with more than 1 service] LPS-177328 Refactor DTOConverter components with more than 1 service type from headless-commerce-admin-pricing-impl part 1

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-177328](https://issues.liferay.com/browse/LPS-177328)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
@RenanUmV No regressions found.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/resource/v1_0/DiscountAccountGroupResourceImpl.java [hunk 1]
```diff
@@ -204,8 +204,12 @@ private List<DiscountAccountGroup> _toDiscountAccountGroups(
 	@Reference
 	private CommerceDiscountService _commerceDiscountService;
 
-	@Reference
-	private DiscountAccountGroupDTOConverter _discountAccountGroupDTOConverter;
+	@Reference(
+		target = "(component.name=com.liferay.headless.commerce.admin.pricing.internal.dto.v2_0.converter.DiscountAccountGroupDTOConverter)"
+	)
```
Comment (rafaprax) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/resource/v1_0/DiscountAccountGroupResourceImpl.java:209]: Hey @RenanUmV, you used the wrong component.name 😢. Please break this commit in two, one for each DTO version. It'll avoid mistakes like this one. Thx

PS: please apply to other commits in this pr

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
@RenanUmV There is an unsatisfied reference error on startup
```
[beanshell] 2023-03-24 01:06:01.051 WARN  [main][SystemCheckOSGiCommands:154] Declarative Service Unsatisfied Component Checker check result: 
[beanshell] Bundle {id: 1300, name: com.liferay.headless.commerce.admin.pricing.impl, version: 4.0.58}
[beanshell] 	Declarative Service {id: 5924, name: com.liferay.headless.commerce.admin.pricing.internal.graphql.servlet.v1_0.ServletDataImpl, unsatisfied references: 
[beanshell] 		{name: _discountAccountGroupResourceComponentServiceObjects, target: null}
[beanshell] 	}
[beanshell] 	Declarative Service {id: 5960, name: com.liferay.headless.commerce.admin.pricing.internal.resource.v1_0.factory.DiscountAccountGroupResourceFactoryImpl, unsatisfied references: 
[beanshell] 		{name: _componentServiceObjects, target: null}
[beanshell] 	}
[beanshell] 	Declarative Service {id: 5950, name: com.liferay.headless.commerce.admin.pricing.internal.resource.v1_0.DiscountAccountGroupResourceImpl, unsatisfied references: 
[beanshell] 		{name: _discountAccountGroupDTOConverter, target: (component.name=com.liferay.headless.commerce.admin.pricing.internal.resource.v1_0.DiscountAccountGroupResourceImpl)}
[beanshell] 	}
```
[Log](https://testray.liferay.com/reports/production/logs/2023-03/test-1-29/test-portal-acceptance-pullrequest(master)/6785/functional-smoke-tomcat90-hypersonic20-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:commerce

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
No regressions found.

**Review by rafaprax — APPROVED** [review]

**Comment by tinatian** [comment]
ci:forward

**Comment by tinatian** [comment]
ci:forward

**Pull request closed without merging**